### ISSUE TITLE
Supabase UI: Change column order

### ISSUE THREAD

**Issue by MentalGear** [supabase/supabase#13686]
# Bug report

Using right-click on a table name, shows the table fields in rows. I seem to remember that it was possible to sort these fields by drag & drop (which also changed the order these fields were shown in new row UI). But this function seems to be gone for the past week or so.

**Comment by MildTomato** [comment]
Closing this issue for now @MentalGear - feel free to re-open if the above is wrong.
We may have also fixed the issue since you posted this issue ⚡️

**Comment by MentalGear** [comment]
No, the issue is sitll present. I was referring to "Edit Table" (image) which opens a view of fields and defines how fields are sorted in the Insert -> Row section.

<img width="302" alt="image" src="https://user-images.githubusercontent.com/2837147/232449133-791a550b-3ef9-4d3f-a5ee-406b8d5967ce.png">

**Comment by plgingras88** [comment]
I agree, this is a little awkward to allow re-ordering only in the Table editor section, especially with a very long vertical scroll caused by many columns. The drag and drop with a very long vertical scroll is a pain and, anyway, does not re-order the real Database related table. I understand postgresql does not allow such feature natively and we could create a view as a work-around, but I believe adding a small abstraction over the GUI on supabase's end should be a decent feature to be considered.

**Comment by BProvDev** [comment]
We agree 200%!
It's a nightmare not being able to save the grid layout at the GUI level AFTER a manual re-arrangement of columns, because Postgres restores the internal order. It's a huge waste of time and we have no choice but to recreate dozens of tables just for that reason, since building Views (as a work around) would be irrational: Each table would have a parallel cloned type of View.

**Comment by kaleidoKurt** [comment]
It’s a major pain that we can’t actually reorder columns in Supabase. The Table Editor lets us drag and drop, but it doesn’t persist after a refresh, and PostgreSQL still enforces the original order. If we want to truly reorder columns, we have to go through the nightmare of recreating tables and migrating data manually, which is overkill for such a simple task.

A “Save Layout” button in the UI would at least make this manageable, but ideally, Supabase should offer a built-in way to reorder columns at the schema level without requiring us to rebuild entire tables. This issue has been raised multiple times—any plans to address it?

**Comment by titusdecali** [comment]
Or, perhaps it should just stay inline with the order described in a Prisma schema? That would be an acceptable solution, at least in my use case with Prisma. We need a better UI for this, definitely.

**Comment by jesse-mylo** [comment]
Can we please consider re-opening this? Regardless of the original post and how it perhaps was nebulously described, the real issue persists and most of the comments refer correctly to it: we need a way to PERMANENTLY change the column order of tables post-create -- i.e. when adding add'l columns to existing tables -- w/o having to do a full migration if at all possible.

Maybe this is a Postgres limitation, I don't know the exact details well enough. 

This isn't a UI issue; this is, for at least one example, about the many many times when you want to modify table columns and put said new columns in a particular spot ... not just "at the end". That said, simply allowing the same "drag and drop" within Table Edit as in Table Create is a logical place for the UI part.

Thanks,
- Jesse

**Comment by VisionaryOS** [comment]
Agreed, I manually reorder them for optimal DB reading - but my team mates don't see my organised table columns...

**Comment by Fruida** [comment]
Hello! Agreed, I have same issue from my end. I recently ordered manually the columns in Supabase in the Table Editor. It's a bit tricky and the columns order is not permanent. It would be great having an option to reorder permanently within the "Edit table" tab.

**Comment by IfnotFr** [comment]
Same here for a drag and drop feature in the "Edit Table" panel. It can be a big mess in a rapid prototyping workflow.

**Comment by Velkamhell97** [comment]
Same here, schemas are very likely to change at the beginning, this feature would give more control and flexibility

**Comment by dddominikk** [comment]
Any chance the column reordering in the Edit Table GUI will be reconsidered? I know it's hardly a high-priority issue but schemas change on an occasion, this would be a useful addition.

**Comment by agbush2** [comment]
Upvoting this please

**Comment by hai-der** [comment]
upvote

**Comment by NurzihanReya** [comment]
Upvote

**Comment by awaseem** [comment]
Hi everyone, Ali from the Dashboard team here. 

I just want to provide an update on this issue. I know a lot of people have requested it. Internally we discussed it and I don't think we'll do this anytime soon. The real issue is because of Postgres, we actually show the column order that Postgres ranks and reordering columns is just not supported right now natively 

To do this, we believe that you need to create a meta-layer on top of Postgres that can store the order of the columns and what you're seeing right now is only stored in local storage. So it will never persist outside of your logged in session, so you can't share the column order to other members. 

Giving you that update, I'm sorry that we're not going to be able to do this in the near future, but if anything changes on my end, I will reopen this issue and let you folks know how we proceed.

**Comment by agbush2** [comment]
Thank you for the honesty.  I just wanted to (re) validate the usecases

Im looking to reorder the columns in my table definition

Ultimately I think this would be a multiple step process

Reorder columns (UX)
Create new table with right order
Copy data
Copy any other stuff (triggers policies etc)
Drop old table
Rename

I think the question is if/how supabase can make this process as seamless as possible to end users as a “managed operation” understanding there may be some caveats (ie the data is too large)

Thanks again for a wonderful product.

A




________________________________

**Comment by grantdiamond** [comment]
how about just being able to sort them alphabetically in the UI? that seems like a good compromise between your internal resources and our desire to quickly find columns

**Comment by jamesAmckinnon** [comment]
I really don't understand why this has not been implemented yet. It is a huge UX flaw in a core feature of the dashboard... the table columns start off in some initial column order state and I can rearrange them to change this state, why can't you store this updated state somewhere for every table and then apply the exact same functionality that would be applied when dragging columns in order to transition to the saved column state every time the table is loaded?

**Comment by its-brody** [comment]
Pretty disappointing to see this was reported 3 years ago and it never got fixed
